src/core/runtime/context.h

```diff
@@ -107,11 +109,18 @@ class LibraryContext {
 // of the Legion API.
 class TaskContext {
  public:
+  TaskContext() = default;
+
   TaskContext(const Legion::Task* task,
               const std::vector<Legion::PhysicalRegion>& regions,
               Legion::Context context,
               Legion::Runtime* runtime);
 
+  TaskContext(const Legion::Task* task, const std::vector<Legion::PhysicalRegion> regions)
```

Comment:
Why is this constructor necessary?

Reply:
Each sub-op is called like this  (via function-pointer):
```
for (int i=0; i<fused_op_len; i++){
      auto descriptor = Core::gpuDescriptors.find(opIDs[i])->second;
      descriptor(context);
}
```
However, the `context` above is not the same as the parent tasks's `context`; Whereas the parent task's context contains the "union" of all `input` `outputs` `reductions` and `scalars` over all subops, the above context should only include the `inputs`, `outputs` etc for sub-operation/task `i`. I used this constructor to build the operation-specific contexts, for holding the specific set of `inputs` `outputs` etc. 

This brings up a potential issue of that the new context's `task` and `regions` fields are accordingly not specific to the sub-op. I'm not sure how much of an issue that is, as I only need the context structure to hold stores and scalars for the sub-ops. Ideally I wanted to avoid any performance costs of creating of new Task objects in the Fused Task, so I elected to keep the new `context` object as such.

Reply:
You do need a new context in this case, but my question is more on why you couldn't use the existing constructor. The constructor you added leaves the runtime and context member uninitialized, leading to a crash if the child task tries to access the runtime (though no task in cunumeric is actually doing that). I think the context you are creating here should still have a valid runtime and context pointer.

That said, I think you actually need a slightly different change here, which allows TaskContext to override the legion task's id. The `task` object here is of the fusion task, so if you pass it as-is to a child task, the child task may think that it's not what it thinks it is. I believe no cuNumeric task is doing actually doing it, but leaving this unaddressed can be a foot gun later on.

Reply:
Yeah it's a valid point. I initially tried using the normal constructor and got runtime errors with from the task deserializer. Recreating the issue I get: ( the error is from span.h : `Assertion pos < size_ failed.`)
```
legion_python: /home/shiv1/fmultiGPU/legate.core/src/core/utilities/span.h:38: decltype(auto) legate::Span<T>::operator[](size_t) [with T = const Legion::PhysicalRegion; size_t = long unsigned int]: Assertion `pos < size_' failed.
Signal 6 received by node 0, process 3131805 (thread 7f8d47bf3000) - obtaining backtrace
Signal 6 received by process 3131805 (thread 7f8d47bf3000) at: stack trace: 19 frames
  [0] = /lib/x86_64-linux-gnu/libpthread.so.0(+0x153c0) [0x7f8d70ce33c0]
  [1] = /lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb) [0x7f8d707f718b]
  [2] = /lib/x86_64-linux-gnu/libc.so.6(abort+0x12b) [0x7f8d707d6859]
  [3] = /lib/x86_64-linux-gnu/libc.so.6(+0x25729) [0x7f8d707d6729]
  [4] = /lib/x86_64-linux-gnu/libc.so.6(+0x36f36) [0x7f8d707e7f36]
  [5] = /home/shiv1/fmultiGPU/legate.core/install/lib/liblgcore.so(+0x5a95d) [0x7f8d4428595d]
  [6] = /home/shiv1/fmultiGPU/legate.core/install/lib/liblgcore.so(legate::TaskDeserializer::_unpack(legate::Store&)+0x10e) [0x7f8d44286d9e]
  [7] = /home/shiv1/fmultiGPU/legate.core/install/lib/liblgcore.so(legate::TaskContext::TaskContext(Legion::Task const*, std::vector<Legion::PhysicalRegion, std::allocator<Legion::PhysicalRegion> > const&, Legion::Internal::TaskContext*, Legion::Runtime*)+0x7f6) [0x7f8d44278626]
  [8] = /home/shiv1/fmultiGPU/legate.core/install/lib/libcunumeric.so(cunumeric::FusedOpTask::gpu_variant(legate::TaskContext&)+0x6e2) [0x7f8d1ce0a852]
```

Reply:
at the time I didn't spend much time debugging this and just wanted something that worked for the time being, but can dive a little deeper into what's causing this

Reply:
I suspect the crash is due to the fusion metadata you added that the current serializer doesn't understand, so this issue is tied with the fusion awareness introduced in the code base, which I suggest removing.